angular-workspace/projects/ni/nimble-angular/rich-text-mention/nimble-rich-text-mention-base.directive.ts

```diff
@@ -0,0 +1,42 @@
+import { Directive, ElementRef, EventEmitter, HostListener, Input, Output, Renderer2 } from '@angular/core';
+import type { RichTextMentionValidity } from '@ni/nimble-components/dist/esm/rich-text-mention/base/models/mention-validator';
+import type { RichTextMention } from '@ni/nimble-components/dist/esm/rich-text-mention/base';
+
+export type { RichTextMentionValidity };
+
+/**
+ * Base class for rich text mention directives
+ */
+@Directive()
+export class NimbleRichTextMentionDirective<T extends RichTextMention> {
+    @Output() public mentionUpdateEvent = new EventEmitter<boolean>();
+
+    @Input() public set pattern(value: string | undefined) {
+        this.renderer.setProperty(this.elementRef.nativeElement, 'pattern', value);
+    }
+
+    public get pattern(): string | undefined {
+        return this.elementRef.nativeElement.pattern;
+    }
+
+    public constructor(private readonly renderer: Renderer2, private readonly elementRef: ElementRef<T>) { }
+
+    public checkValidity(): boolean {
+        return this.elementRef.nativeElement.checkValidity();
+    }
+
+    public getMentionedHrefs(): string[] {
+        return this.elementRef.nativeElement.getMentionedHrefs();
+    }
+
+    public get validity(): RichTextMentionValidity {
+        return this.elementRef.nativeElement.validity;
+    }
+
+    @HostListener('mention-update', ['$event'])
+    public onMentionUpdate($event: Event): void {
+        if ($event.target === this.elementRef.nativeElement) {
+            this.mentionUpdateEvent.emit();
+        }
+    }
```

Comment:
Doesn't mention-update include some data for what is being typed? An object like `{filter: string}`. Seems weird to not be symmetric across the boundary.

But I also think we established a pattern of not doing `@Output` definitions and just relying on the event propagating on it's own if we aren't trying to setup two-way binding. @mollykreis do you remember what our guidelines are?

Reply:
Don't add the `@Output`. Just keep the exported `MentionUpdateEventDetail`.
Angular apps that want to listen for the event from the component can bind to the `mention-update` event that is fired from the component directly. We don't need to define `@Output` for it.

Reply:
Removed `@HostListener` and `EventEmitter` for mention-update event.